### PR TITLE
Direction finding connectionless locator sample app: add support for nRF5340

### PR DESCRIPTION
**Pull request by ppryga**
Update configuration to allow build of the sample application with nRF5340.
Add information about nRF5340 support in README.rst.

This PR requires following other PRs to be merged:

- https://github.com/zephyrproject-rtos/zephyr/pull/36047 to be merged in upstream and cherry-picked into sdk-zephyr repository.
- https://github.com/nrfconnect/sdk-nrf/pull/4763 to be merged into main.
- Upmerge of sdk-zephyr repository has to be finished.

### CODE REVIEW

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_SW_SPLIT=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_SYNC_PERIODIC=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf:13]: Remove duplicated options:

CONFIG_BT_CTLR_ADV_EXT=y
CONFIG_BT_CTLR_SYNC_PERIODIC=y

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+CONFIG_BT_EXT_ADV=y
+
+# Required to enable BT_PER_ADV_SYNC_MAX
+CONFIG_BT_PER_ADV_SYNC=y
+CONFIG_BT_OBSERVER=y
+
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_SW_SPLIT=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf:17]: Remove duplicated options:
CONFIG_BT_CTLR_ADV_EXT=y
CONFIG_BT_CTLR_SYNC_PERIODIC=y

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 3]
```diff
@@ -16,7 +16,7 @@ The sample supports the following development kits:
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
-   :rows: nrf52833dk_nrf52833
+   :rows: nrf52833dk_nrf52833, nrf5340dk_nrf5340_cpuapp_and_cpuappns
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:19]: Can we use `nrf5340dk_nrf5340_cpuapp_and_cpuappns`, or should we split this into 2 boards?

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 4]
```diff
@@ -38,6 +38,15 @@ Configuration
 
 |config|
 
+This sample configuration is split into the following two files:
+
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
+
+Board specific configuration involves configuring the Bluetooth LE controller configuration.
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:46]: `Board specific configuration involves configuring the Bluetooth LE controller`

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 3 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:19]: `nrf5340dk_nrf5340_cpuapp_and_cpuappns` is used by other BLE samples in nrf-sdk, so I assume that this it the correct one.

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 4 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:46]: Removed 'controller' word.

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf (hunk 2 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf:17]: Removed

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 4 above)
Comment (mia-ko) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:46]: Why remove the 'controller' word?
I think this sentence should be: Board-specific configuration involves configuring the Bluetooth LE controller.
Right?

Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:46]: @mia-ko I've put wrong word in comment. The removed word is ` configuration`.

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Reguired to enable BT_BUF_CMD_TX_SIZE for LE Set Extended Advertising Data command
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf:7]: Required

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+&radio {
+	status = "okay";
+	/* This is a number of antennas that are available on antenna matrix
+	 * designed by Nordic. For more information see README.rst.
+	 */
+	dfe-antenna-num = < 12 >;
+	/* This is a setting that enables antenna 12 (in antenna matrix designed
+	 * by Nordic) for Tx PDU. For more information see README.rst.
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay:14]: RX ?

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+	 */
+	dfe-pdu-antenna = <0x0>;
+
+	/* These are GPIO pin numbers that are provided to
+	 * Radio peripheral. The pins will be acquired by Radio to
+	 * drive antenna switching when AoD is enabled.
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay:20]: AoA ?

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 8]
```diff
@@ -46,6 +55,8 @@ To build this sample with angle of departure mode only, set ``OVERLAY_CONFIG`` t
 See :ref:`cmake_options` for instructions on how to add this option.
 For more information about using configuration overlay files, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
 
+To build this sample for :ref:`nRF5340 DK <ug_nrf5340>`, with angle of arrival mode only, add content of :file:`overlay-aoad.conf` file to :file:`child_image/hci_rpmsg.conf` file.
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:58]: `overlay-aod`

**Review by MarekPieta — APPROVED** [review]

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 9]
```diff
@@ -38,6 +38,15 @@ Configuration
 
 |config|
 
+This sample configuration is split into the following two files:
+
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
```
Comment (mia-ko) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:44]:
```suggestion
* board specific configuration is available in :file:`boards/<BOARD>.conf` file
```

**Review by mia-ko — CHANGES_REQUESTED** [review]
Just one typo left! Otherwise it looks good :)

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 8 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:58]: Corrected, thank you.

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay (hunk 7 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay:20]: Corrected, thank you.

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay (hunk 6 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay:14]: Corrected, thank you.

File: samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf (hunk 5 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf:7]: Corrected, thank you.

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 9 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:44]: Corrected, thank you.

**Comment by ppryga** [comment]
PR converted to Draft on request from @thst-nordic due to limited CI resources. Reviewing and approval process still continues without change.

**Review by mia-ko — APPROVED** [review]

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_SW_SPLIT=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_SYNC_PERIODIC=y
+
+# Enable Direction Finding Feature including AoA and AoD
+CONFIG_BT_CTLR_DF=y
+CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=n
```
Comment (KAGA164) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf:15]: Shouldn't that be under line 17 for clarity ?

**Review by KAGA164 — CHANGES_REQUESTED** [review]

**Comment by ppryga** [comment]
Moved DTS overlay from `child_image/hci_rpmsg.overlay` to `child_image/hci_rpmsg/nrf5340dk_nrf5340_cpunet.overlay`.
Corrected antenna switches GPIO mapping in the DTS overlay.

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf (hunk 10 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf]: Moved below line `# Enable Direction Finding Feature including AoA and AoD`

**Comment by ppryga** [comment]
@KAGA164 I've provided requested change. Take a look on PR, please.

**Review by KAGA164 — APPROVED** [review]

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
> This PR requires following other PRs to be merged:
> [...]
> [nrf fromtree] Add support for nRF5340 in direction finding sdk-zephyr#578 to be merget into sdk-zephyr.
> [nrf fromtree] Add support for nRF52820 in direction finding sdk-zephyr#577 to be merget into sdk-zephyr.

Please consolidate the two sdk-zephyr PRs into a single PR, and pull them into this PR's west.yml, so we can get CI results.

**Review by mbolivar-nordic — APPROVED** [review]
Changes look good, thanks for bringing in the manifest update.

**Comment by ppryga** [comment]
Rebased again due to issues in CI jobs.

**Comment by mbolivar-nordic** [comment]
This will have to wait to be merged until after the upmerge I just started.

**Comment by ppryga** [comment]
Removed commit that updated west manifest.

**Pull request merged**